File: leet-code/educative.io/13-top-k-elements/kth-largest-element-in-an-array.py
# https://leetcode.com/problems/kth-largest-element-in-an-array/

from typing import List
import heapq

class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        heapq._heapify_max(nums)
        largestKth = None
        
        i = 0
        while len(nums) > 0 and i < k:
            # if the popped number is kth index, it is the largest
            popped = heapq._heappop_max(nums)
            if i == k - 1:
                largestKth = popped

            i += 1
    
        return largestKth

File: coding-challenges/interviews.school/04-sorting/kth-largest-element-in-an-array.cpp
// https://leetcode.com/problems/kth-largest-element-in-an-array/

#include <algorithm>
#include <vector>

class Solution {
public:
    int findKthLargest(std::vector<int>& nums, int k) {
        std::sort(nums.begin(), nums.end());
        return nums[nums.size() - k];
    }
};
